### PR TITLE
Add new docker build script and copy build_ubuntu script in docker di…

### PR DESCRIPTION
**Pull request by agwhittle**
…rectory.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -1,19 +1,81 @@
-# To build the Docker container, run the following lines from your proteus root directory:
-# cd docker
-# docker build --rm -t proteus:latest .
```
Comment (lukethehuman) [inline docker/Dockerfile:3]: Do we want to remove this comment? It looks useful.

File: docker/Dockerfile [hunk 2]
```diff
@@ -1,19 +1,81 @@
...
 
 # Build Proteus
-RUN cd /$WORKDIR && \
-    git clone https://github.com/aurora-multiphysics/proteus.git && \
-    cd proteus && \
-    make -j"$compile_cores"
-
-# Run tests
-RUN cd /proteus && \
-    ./run_tests
```
Comment (lukethehuman) [inline docker/Dockerfile:19]: Is there a reason for removing the tests? Is this to reduce build time?

File: docker/Dockerfile [hunk 3]
```diff
@@ -1,19 +1,81 @@
...
-# cd docker
-# docker build --rm -t proteus:latest .
+# Use Ubuntu as the base image
+FROM ubuntu:22.04
 
-# Get latest base MOOSE image
-FROM idaholab/moose:latest
+# Set environment variables
+ENV DEBIAN_FRONTEND=noninteractive
+ENV PROTEUS_DIR=/proteus
```
Comment (lukethehuman) [inline docker/Dockerfile:6]: The proteus installation is in `/proteus` but the working directory we enter at the end is `/opt/proteus` (see ln 77). This is confusing. The user experience is currently to enter a directory called proteus, run `ls` and be faced with an empty directory. Then the user may find by moving through the directory structure that the actual installation is in another directory called proteus in a different location. Is this a typo and both are meant to be `/opt/proteus`?

**Review by lukethehuman — CHANGES_REQUESTED** [review]
The docker container builds successfully, great work. However, I have left a few comments for changes I'm unsure about. Mainly, either there has been a typo, or we need to make the directory structure more user friendly.

**Comment by agwhittle** [comment]
All issues with Dockerfile should be fixed.

**Review by lukethehuman — APPROVED** [review]
All comments from last review addressed. Branch is ready for merge.

**Pull request merged**